servicetalk-buffer-netty/src/main/java/io/servicetalk/buffer/netty/UnreleasableReadOnlyByteBufferBuf.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright © 2018-2021 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.buffer.netty;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+
+import java.nio.ByteBuffer;
+
+public class UnreleasableReadOnlyByteBufferBuf extends ReadOnlyByteBufferBuf {
```

Comment:
I understood your motivation to keep the copied class as close as possible to the original version from netty, but I don't think it's necessary and we do not really follow this pattern. Consider using only one class - `UnreleasableReadOnlyByteBufferBuf` (by merging it together with `ReadOnlyByteBufferBuf`) and making it `final` + pkg-private.

Reply:
Thanks for the clarification. I was hesitant for a while to keep the separation of concerns, but in our case it actually makes more sense to merge these classes the way you describe.